app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -614,34 +522,177 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
+            }
+        }
 
-        //Ugh, copied from the odkview mostly, that's stupid
-        for(QuestionWidget q : ((ODKView)mCurrentView).getWidgets()) {
-            //Figure out if we have a pending intent widget
-            if (q instanceof IntentWidget) {
-                if(((IntentWidget) q).getFormId().equals(mFormController.getPendingCalloutFormIndex())) {
-                    bestMatch = (IntentWidget)q;
-                }
+        File fileToReturn;
+        if (!savedScaledImage) {
+            // If we didn't create a scaled image and save it to the final path, then relocate the
+            // original image from the temp filepath to our final path
+            File finalFile = new File(finalFilePath);
+            if (!originalImage.renameTo(finalFile)) {
+                throw new IOException("Failed to rename " + originalImage.getAbsolutePath() +
+                        " to " + finalFile.getAbsolutePath());
+            } else {
+                fileToReturn = finalFile;
+            }
+        } else {
+            // Otherwise, relocate the original image to a raw/ folder, so that we still have access
+            // to the unmodified version
+            String rawDirPath = instanceFolder + "/raw";
+            File rawDir = new File(rawDirPath);
+            if (!rawDir.exists()) {
+                rawDir.mkdir();
+            }
+            File rawImageFile = new File (rawDirPath + "/" + imageFilename);
+            if (!originalImage.renameTo(rawImageFile)) {
+                throw new IOException("Failed to rename " + originalImage.getAbsolutePath() +
+                        " to " + rawImageFile.getAbsolutePath());
+            } else {
+                fileToReturn = rawImageFile;
             }
         }
-        return bestMatch;
+        return fileToReturn;
+    }
+
+
+    /**
+     * Processes the return from an image capture intent, launched by either an ImageWidget or
+     * SignatureWidget
+     *
+     * @param isImage true if this was from an ImageWidget, false if it was a SignatureWidget
+     */
+    private void processCaptureResponse(boolean isImage) {
+        /* We saved the image to the tempfile_path, but we really want it to be in:
+         * /sdcard/odk/instances/[current instance]/something.jpg so we move it there before
+         * inserting it into the content provider. Once the android image capture bug gets
+         * fixed, (read, we move on from Android 1.6) we want to handle images the audio and
+         * video
+         */
+
+        // The intent is empty, but we know we saved the image to the temp file
+        File originalImage = ImageWidget.TEMP_FILE_FOR_IMAGE_CAPTURE;
+        try {
+            File unscaledFinalImage = moveAndScaleImage(originalImage, isImage);
+            saveImageWidgetAnswer(unscaledFinalImage);
+        } catch (IOException e) {
+            showCustomToast(Localization.get("image.capture.not.saved"), Toast.LENGTH_LONG);
```

Comment:
We set some text in the e above, we should likely log or throw that error so we know what he issue was on a console, rather than losing the info about what was in 'e'